### ISSUE TITLE
[iOS][Audio] Can't play an audio when the app is already in background

### ISSUE THREAD

**Issue by gsantiago** [expo/expo#5472]
### Environment
```
  Expo CLI 2.19.5 environment info:
    System:
      OS: Linux 5.0 Ubuntu 19.04 (Disco Dingo)
      Shell: 3.0.2 - /usr/bin/fish
    Binaries:
      Node: 10.15.2 - /tmp/yarn--1567012357926-0.5700793194663918/node
      Yarn: 1.16.0 - /tmp/yarn--1567012357926-0.5700793194663918/yarn
      npm: 5.8.0 - /usr/bin/npm
    npmPackages:
      @types/react: ^16.8.19 => 16.8.19 
      @types/react-native: ^0.57.60 => 0.57.62 
      expo: ^34.0.3 => 34.0.4 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz => 0.59.8 
    npmGlobalPackages:
      expo-cli: 3.0.10
```

This bug only happens in iOS.

### Steps to Reproduce

- Try to play a music when the app is in background. You can check the code in my snack: https://snack.expo.io/@gsantiago/reproducible-bug---failed-to-play-audio-when-the-app-is-in-background

### Expected Behavior

- The app should play the audio when it's already in background.

### Actual Behavior

- In Android, it works perfectly fine.
- In iOS, I get the following error:

```
[Error: Play encountered an error: audio session not activated.]
```

### Reproducible Demo

https://snack.expo.io/@gsantiago/reproducible-bug---failed-to-play-audio-when-the-app-is-in-background

And, yes. I have set `staysActiveInBackground: true` and all other necessary modes to play in background, and also UBackgroundModes: ['audio'] in the app.json.

**Comment by vikash-gupta** [comment]
Hello @gsantiago , I also faced the same issue that you mentioned. The only workaround I found is to keep the screen awake pro-grammatically, however as soon as user locks the screen, next audio doesn't start. 

Did you find any workaround?

**Comment by gsantiago** [comment]
@vikash-gupta I have disabled the background audio in iOS for now.

**Comment by vikash-gupta** [comment]
Bumping this up... Can anyone please look into it?

**Comment by alexandermckay** [comment]
Would love to see this resolved.

**Comment by spdaly** [comment]
+1

**Comment by AlexisChup** [comment]
+1

**Comment by AlexisChup** [comment]
My entire App is about sounds which start when the app is in Background, so it would be wonderful if this feature worked !

**Comment by wanghoppe** [comment]
My app is a music player... Now you can only play one song in the background. Looking forward updates

**Comment by Lyubomyr** [comment]
Is any progress on this issue?

**Comment by bbbbighead** [comment]
same issue encountered

**Comment by dani-mp** [comment]
I currently use this library in production (and for a long time now) and I can't reproduce this bug. I'm super interested in knowing what's the real problem so I would appreciate if someone that can reproduce it shares a minimal project with a description of the steps to take to make it fail. If so, I will take a look at it.

**Comment by Lyubomyr** [comment]
My steps to reproduce:
1. Use the last version of library. 
2. Use background options for sound:
    Audio.setAudioModeAsync({
      allowsRecordingIOS: false,
      staysActiveInBackground: true,
      interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
      playsInSilentModeIOS: true,
      shouldDuckAndroid: true,
      interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX,
      playThroughEarpieceAndroid: true,
      useNativeControls: false
    })
3. Play some test video file using the Video from expo-av.
4. Put application to background.
Result: audio from video file continue work only on Android, on IOS production, IOS expo app or simulator no sound at all. 

4. Redo the same using the same video file but now Audio from expo-av

**Comment by peterkortvel** [comment]
same issue

**Comment by gwendall** [comment]
+1, any news ?

**Comment by ohsik** [comment]
+1 same issue

**Comment by AwesomeZaidi** [comment]
This is a huge issue

**Comment by AwesomeZaidi** [comment]
iAny other known audio players that can handle this functionality in React Native?

**Comment by AwesomeZaidi** [comment]
@sjchmiela @cruzach any thoughts here?

**Comment by brentvatne** [comment]
from our [managed workflow limitations docs page](https://docs.expo.io/introduction/why-not-expo/)

<img width="1082" alt="Screen Shot 2020-05-30 at 3 34 52 PM" src="https://user-images.githubusercontent.com/90494/83340295-33b6e700-a28b-11ea-847c-179f5114ccb9.png">

you may need to eject if you want more complete background audio functionality, for now

**Comment by ohsik** [comment]
@brentvatne Wondering if there's a plan to work on this in the near future?

**Comment by brentvatne** [comment]
i wish we had time to but are pretty busy at the moment with bigger infrastructure improvements because that work will unblock users from solving issues like this for themselves.

if anyone here would like to hire someone to work on this feature for you, i would recommend reaching out to https://swmansion.com/ - we are partners with them and they have a lot of experience at the org with expo and react-native. i would highly recommend them

**Comment by ohsik** [comment]
@brentvatne Thank you for the response. Hope to see that update soon!

**Comment by drsounds** [comment]
Expo should implement an audio playback service in the native expo runtime, eg run not on react native thread which handles playback in bakgrund per default which is communicated over bridge instead of running audio on the ui layer (as now with expo av)

**Comment by IjzerenHein** [comment]
As far as I'm aware, this was due to an issue in expo-av (similar issue was reported here https://github.com/expo/expo/pull/9363)

You should now be able to resume/play audio when in the background when `staysActiveInBackground=true` is set.

This has not yet landed in the Expo client, but you can try it out in a bare project using:

`yarn install expo-av@8.4.0`

You can find the changelog here: https://github.com/expo/expo/blob/master/packages/expo-av/CHANGELOG.md

Let us know whether that fixes it for you.

**Comment by sgvolpe1** [comment]
Same issue

**Comment by Jonnathangte68** [comment]
same issue. What's the status of this? 👀

**Comment by ide** [comment]
This closed issue is two years old and new issues with similar symptoms may have different root causes. The best way to help address a problem is to suss out the cause and write it up in a new issue.

**Comment by titushartm** [comment]
Im facing a simular problem with "expo-av": "^13.2.1". I try to record audio with my app. When interruption by locking the screen the recording stops. The recording time is still going and after unlocking the screen the recording continues.

UIBackgroundModes: ["audio"]

```
await Audio.setAudioModeAsync({
            allowsRecordingIOS: true,
            staysActiveInBackground: true,
            interruptionModeIOS: InterruptionModeIOS.DuckOthers,
            playsInSilentModeIOS: true,
            shouldDuckAndroid: true,
            interruptionModeAndroid: InterruptionModeAndroid.DuckOthers,
            playThroughEarpieceAndroid: false,
          });
```

**Comment by kiorq** [comment]
For those who the background audio works in the background, you may want to check if UIBackgroundModes in both the app.json and Info.plist file. I believe once you have a bare app (you have an `./ios` dir) you'll need to update the Info.plist manually as well.
